### PR TITLE
Add support for subscription and resource group level msi scopes

### PR DESCRIPTION
**Pull request by jjcollinge**
Addresses: #28

I've purposefully not used a regex to parse the subscription id (guid) directly and instead used the package github.com/beevik/guid to keep this code as simple as possible. If we'd rather not rely on that package then let me know and I'll update the regex to cover the more complex case.

### CODE REVIEW

File: pkg/driver/aci-driver.go [hunk 1]
```diff
@@ -316,6 +329,45 @@ func (d *aciDriver) processConfiguration(config map[string]string) error {
...
+// * /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/...
+func validateMSIScope(scope string) error {
+	parts := strings.Split(scope, "/") // Leading slash '/' in scope adds an empty part.
+	if len(parts) < 3 {
+		return errors.New("invalid msi scope, scope must start with /subscriptions/<subscriptionID>")
+	}
+
+	// Azure subscription ids should be represented as 32 bit guids.
+	subID := parts[2]
+	if _, err := guid.ParseString(subID); err != nil {
```
Comment (simongdavies) [inline pkg/driver/aci-driver.go:344]: is there a reason that you cannot use [github.com/google/uuid](https://github.com/google/uuid/blob/512b657a42880af87e9f0d863aa6dccf3540d4ba/uuid.go) for this seen as this is already imported?

**Review by simongdavies — CHANGES_REQUESTED** [review]

File: pkg/driver/aci-driver.go (hunk 1 above)
Comment (jjcollinge) [reply pkg/driver/aci-driver.go]: 🤦‍♂️Good spot - I'm deep in the C# world atm and was looking for Guid parsing and completely forgot about uuid! It should be a drop in replacement so I'll update now.

**Review by simongdavies — APPROVED** [review]

**Comment by simongdavies** [comment]
@jjcollinge can you sign your commit please?

**Review by simongdavies — CHANGES_REQUESTED** [review]
@jjcollinge  can you sign all the commits please? (or you can squash)

**Review by simongdavies — APPROVED** [review]

**Comment by jjcollinge** [comment]
Sorry @simongdavies I missed the notification about your comment - I'm happy to sign the commits but I won't be able to until Friday if that works.

**Comment by jjcollinge** [comment]
@simongdavies sorry about the delay I've rebased and signed the commits now - let me know if you need anything else.

**Pull request merged**